src/classes/CON_ContactMerge_TDTM.cls

```diff
@@ -105,93 +112,253 @@ public class CON_ContactMerge_TDTM extends TDTM_Runnable {
+         * @description Fixes records involved in the merge process
+         * @param mergeKeys - the ids releveant to the merge process
+         */
+        private void fixup(List<String> mergeKeys) {
+            setMergeKeyIds(mergeKeys);
+
+            List<Account> losersAccounts = getAccounts(losersAccountIds);
+            List<Contact> winningContacts = getContacts(winningContactIds);
+
+            householdAcctIds = getHouseholdAcctIds(losersAccounts);
+            householdAcctIds.addAll(getHouseholdAcctIds(winningContacts));
 
-        // extract out our Contact and Account Id's from the mergeKeys
-        for (string mergeKey : listMergeKey) {
-            listContactIdLoser.add(idFromMergeKey(mergeKey, 0));
-            listAccountIdLoser.add(idFromMergeKey(mergeKey, 1));
-            listContactIdWinner.add(idFromMergeKey(mergeKey, 2));
+            oldAcctIdForOppsByWinningContactId = getOldAcctIdForOppsByWinningContactId(losersAccounts);
+            newAcctIdForOppsByWinningContactId = getNewAcctIdForOppsByWinningContactId(winningContacts);
+
+            fixupAccountData();
+            fixupContactData(winningContacts);
         }
-    
-        // get the winning Contacts (with the fields we need)
-        RLLP_OppRollup_UTIL oppRollupUtil = new RLLP_OppRollup_UTIL(); 
-        string strSoql = oppRollupUtil.buildContactQuery();
-        strSoql = strSoql.replace('from Contact', '');
-        strSoql += ', AccountId, Account.npe01__SYSTEM_AccountType__c, Primary_Affiliation__c from Contact where id IN :listContactIdWinner';
-        list<Contact> listContactWinner = Database.query(strSoql);
-
-        // create maps of contactId's to AccountId's that some of our routines need.
-        map<Id,Id> mapContactIdAccountIdWinner = new map<Id,Id>();
-        map<Id,Id> mapContactIdAccountIdLoser = new map<Id,Id>();
-        list<Id> listAccIdHH= new list<Id>();
-        for (Contact con : listContactWinner) {
-        	if (con.Account.npe01__SYSTEM_AccountType__c == CAO_Constants.HH_ACCOUNT_TYPE)
-        	   listAccIdHH.add(con.AccountId);
-            if (con.Account.npe01__SYSTEM_AccountType__c == CAO_Constants.HH_ACCOUNT_TYPE ||
-                con.Account.npe01__SYSTEM_AccountType__c == CAO_Constants.ONE_TO_ONE_ORGANIZATION_TYPE ||
-                con.Account.npe01__SYSTEM_AccountType__c == CAO_Constants.BUCKET_ORGANIZATION_TYPE) {
-                mapContactIdAccountIdWinner.put(con.Id, con.AccountId);        
+
+        /*******************************************************************************************************
+        * @description Returns the account Id of accounts with household account types
+        ********************************************************************************************************/
+        @TestVisible
+        private List<Id> getHouseholdAcctIds(List<Account> accounts) {
```

Comment:
Suggestion: `getHouseholdIds() ` or `extractHouseholdIds()`? Usually `get` is reserved to indicate method getting data from database.

Reply:
I went with build on all of the sets and map generators, thanks for the suggestion and please let me know if that reads better now.